### PR TITLE
Feature/create exam redux

### PR DESCRIPTION
**Pull request by ValentinChion**
Some new features in this branch:
- Real header/navbar
- Send exam to API when using instruction component
- Fix dropdown components experience in instruction component

### CODE REVIEW

File: frontend/src/react/App/App.css [hunk 1]
```diff
@@ -1,13 +1,33 @@
 body {
   background: #ECF0F3;
 }
-
+.is-white-important {
+  background: white !important;
+}
 .is-a-link {
-  color: #3273dc;
+  color: #3273dc !important;
```
Comment (NansD) [inline frontend/src/react/App/App.css]: Ils sont indispensables les `!important` ici ?
parce que c'est plutôt #badPractice de faire ça :(

File: frontend/src/react/Instructions/Instructions.js [hunk 2]
```diff
@@ -1,21 +1,26 @@
...
     };
 
     /**
      * trigger the boolean in state with str as input
      * @param str
      * @returns undefined
      */
-    triggerDropdown = (str) => () => {
-        this.setState({[str]: !this.state[str]})
+    triggerDropdown = (str) => (e) => {
```
Comment (NansD) [inline frontend/src/react/Instructions/Instructions.js]: on n'en a plus besoin ?

File: frontend/src/react/Instructions/Instructions.js [hunk 3]
```diff
@@ -26,18 +31,62 @@ export default class Instructions extends Component {
...
+                redirectExercices: true})
+        }
+    };
+
+    triggerActive = (e) => {
+        e.target.className = e.target.className + " is-active";
+    };
+
+    triggerInactive = (e) => {
+        e.target.className = e.target.className.substr(0, e.target.className.length - 10);
```
Comment (NansD) [inline frontend/src/react/Instructions/Instructions.js]: Je propose plutôt : 
`e.target.className = e.taget.className.replace('is-active', '');`

File: frontend/src/repository/exams.js [hunk 4]
```diff
@@ -2,18 +2,18 @@ import axios from 'axios';
 import moment from "moment";
 
 export const getExams = async () => {
-    const { data } = await axios.get('http://localhost:4000/api/exams');
+    const { data } = await axios.get('/api/exams');
```
Comment (bdebever) [inline frontend/src/repository/exams.js]: As mentioned @ValentinChion , if you can revert the changes on this file to avoid conflicts.

File: frontend/src/react/Instructions/Instructions.js [hunk 5]
```diff
@@ -26,18 +22,62 @@ export default class Instructions extends Component {
...
+
+        await this.props.createExam(examData);
+        if (this.props.exams.exam._id) {
+            this.setState({
+                idRedirect: this.props.exams.exam._id,
+                redirectExercices: true})
+        }
+    };
+
+    triggerActive = (e) => {
```
Comment (bdebever) [inline frontend/src/react/Instructions/Instructions.js]: I'd say this function as well as the inactive one should be put inside an helper file of some sort. I'm pretty confident we will have to use that again later!

File: frontend/src/redux/exams/reducers.js [hunk 6]
```diff
@@ -21,13 +24,20 @@ export default combineReducers({
...
         return action.exam;
-      case ACTIONS.SET_POST_EXAM_SUCCESS:
-        return [
+      case ACTIONS.SET_PATCH_EXAMS_SUCCESS:
+        return action.exams;
+        case ACTIONS.SET_POST_EXAM_SUCCESS:
+        return state.exams.length ? {
+            ...state,
+            exam: action.exam,
+            exams: [
```
Comment (bdebever) [inline frontend/src/redux/exams/reducers.js:33]: Can you clarify why we need both `exam` and `exams` property on this state object in the singular post_exam_success?

**Review by bdebever — COMMENTED** [review]
Added my comments inline in the files.

**Review by FelixDelcourt — APPROVED** [review]

File: frontend/src/redux/exams/reducers.js (hunk 6 above)
Comment (ValentinChion) [reply frontend/src/redux/exams/reducers.js:33]: Because otherwise I need to check which is my new exam in the exams array. Here I know that "exam" is the new one and I can send it's params in the route !

File: frontend/src/react/Instructions/Instructions.js (hunk 3 above)
Comment (ValentinChion) [reply frontend/src/react/Instructions/Instructions.js]: Done :rocket:

File: frontend/src/react/Instructions/Instructions.js (hunk 2 above)
Comment (ValentinChion) [reply frontend/src/react/Instructions/Instructions.js]: Deleted :+1:

File: frontend/src/react/App/App.css (hunk 1 above)
Comment (ValentinChion) [reply frontend/src/react/App/App.css]: One is necessary, I've removed the other ones, thx
